addons/account/models/account_move.py

```diff
@@ -310,6 +310,10 @@ def fields_view_get(self, view_id=None, view_type='form', toolbar=False, submenu
     def create(self, vals):
         move = super(AccountMove, self.with_context(mail_create_nolog=True, check_move_validity=False, partner_id=vals.get('partner_id'))).create(vals)
         move.assert_balanced()
+        for move in self:
+            if move.line_ids:
+                if not all([x.company_id.id == move.company_id.id for x in move.line_ids]):
```

Comment:
use `if any` (less computation)

I think it is better to create an @api.contraints(), no ?